code/game/objects/structures.dm

```diff
@@ -16,9 +16,7 @@
 		cameranet.updateVisibility(src)
 
 /obj/structure/blob_act(obj/effect/blob/B)
-	if(!density)
-		qdel(src)
-	if(prob(50))
+	if(density && prob(50))
 		qdel(src)

```

Comment:
This seems wrong, This would spare any nondense structure from any kind of blob destruction where previously they would have been always destroyed.